### PR TITLE
Alpha toast core

### PR DESCRIPTION
**Pull request by Wagnerp**
* Get the basics working

image

### CODE REVIEW

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static class KryptonMessageBox
         /// <param name="contentLinkArea">Specifies the area within the <see cref="KryptonLinkWrapLabel"/> to be regarded as a link. See <see cref="LinkArea"/>.</param>
         /// <param name="messageTextAlignment">Specifies how the message text should be aligned. See <see cref="System.Drawing.ContentAlignment"/> for supported values.</param>
         /// <returns>One of the System.Windows.Forms.DialogResult values.</returns>
-        public static DialogResult Show(string text, string caption, bool? showCtrlCopy = null,
+        public static DialogResult Show(string? text, string caption, bool? showCtrlCopy = null,
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs:42]: How can the first string be nullable, but the caption not be ?

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonThemeComboBox.cs [hunk 2]
```diff
@@ -12,190 +12,218 @@
...
+		/// </summary>
+		[Category(@"Visuals")]
+		[Description(@"Custom Theme to use when `Custom` is selected")]
+		[DefaultValue(null)]
+		public KryptonCustomPaletteBase? KryptonCustomPalette { get; set; }
+
+		//[EditorBrowsable(EditorBrowsableState.Never)]
+		public KryptonManager Manager
+		{
+			get;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonThemeComboBox.cs:87]: I think this should return the owner form manager before creating a new one

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+                Screen.PrimaryScreen.WorkingArea.Height - Height - 5);
+        }
+
+        private void VisualToastNotificationBasicForm_Load(object sender, EventArgs e)
+        {
+            UpdateLocation();
+
+            ShowCloseButton();
+
+            _timer?.Start();
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:169]: When will the timer be null?
Remove nullable

File: Source/Krypton Components/Krypton.Toolkit/General/Toast Notification/KryptonBasicToastNotificationData.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public struct KryptonBasicToastNotificationData
+    {
+        #region Public
+
+        public bool? ShowCloseBox { get; set; }
+
+        public int? CountDownSeconds { get; set; }
+
+        public string NotificationContent { get; set; }
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/General/Toast Notification/KryptonBasicToastNotificationData.cs:48]: This should be nullable as well, As there might be title only

File: Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs [hunk 5]
```diff
@@ -80,12 +82,39 @@ public class PaletteInputControlContentStates : Storage,
...
+        #region SynchronizeDropDownWidth
+
+        /// <summary>Gets or sets a value indicating whether [synchronize drop down width].</summary>
+        /// <value><c>true</c> if [synchronize drop down width]; otherwise, <c>false</c>.</value>
+        [KryptonPersist(false)]
+        [Category(@"Visuals")]
+        [Description(@"Adjust the drop-down width, based on the controls width.")]
+        [DefaultValue(null)]
+        [AllowNull]
+        public bool SynchronizeDropDownWidth
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs:101]: Need `Should###` and `Reset###` on these, otherwise they show up as always being bold in the designer

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs [hunk 6]
```diff
@@ -69,10 +69,11 @@ private void UpdateFadeValues()
         private void UpdateFonts()
         {
             kwlblContent.StateCommon.Font = _basicToastNotificationData.NotificationContentFont ??
-                                KryptonManager.CurrentGlobalPalette.BaseFont;
+                                            KryptonManager.CurrentGlobalPalette.BaseFont;
 
             kwlblHeader.StateCommon.Font = _basicToastNotificationData.NotificationTitleFont ??
-                               new Font(KryptonManager.CurrentGlobalPalette.BaseFont.FontFamily, 11f, FontStyle.Bold);
+                                           new Font(KryptonManager.CurrentGlobalPalette.BaseFont.FontFamily, 11f,
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:75]: Should use Kryptons "HeaderStyle" font, because 11 might be smaller than the Base font size !

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs (hunk 1 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs:42]: Reverted

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs (hunk 3 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:169]: Removed nullable

File: Source/Krypton Components/Krypton.Toolkit/General/Toast Notification/KryptonBasicToastNotificationData.cs (hunk 4 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/General/Toast Notification/KryptonBasicToastNotificationData.cs:48]: Now nullable

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonThemeComboBox.cs (hunk 2 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonThemeComboBox.cs:87]: `Manager` is now assignable

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs [hunk 7]
```diff
@@ -118,7 +118,7 @@ public static class KryptonMessageBox
         /// <param name="contentLinkArea">Specifies the area within the <see cref="KryptonLinkWrapLabel"/> to be regarded as a link. See <see cref="LinkArea"/>.</param>
         /// <param name="messageTextAlignment">Specifies how the message text should be aligned. See <see cref="System.Drawing.ContentAlignment"/> for supported values.</param>
         /// <returns>One of the System.Windows.Forms.DialogResult values.</returns>
-        public static DialogResult Show(IWin32Window? owner, string? text, string caption, bool? showCtrlCopy = null,
+        public static DialogResult Show(IWin32Window? owner, string text, string caption, bool? showCtrlCopy = null,
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs:121]: Why have these been changed. There is a test for passing just a caption and null for the message ?

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs [hunk 8]
```diff
@@ -53,27 +53,33 @@ public VisualToastNotificationBasicForm(KryptonBasicToastNotificationData data)
 
         private void UpdateText()
         {
-            kwlblContent.Text = _basicToastNotificationData.NotificationContent;
+            kwlblContent.Text = _basicToastNotificationData.NotificationContent ?? null;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:56]: If it's null then its null, why this change ?

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs [hunk 9]
```diff
@@ -53,27 +53,33 @@ public VisualToastNotificationBasicForm(KryptonBasicToastNotificationData data)
...
 
             kwlblHeader.TextAlign =
                 _basicToastNotificationData.NotificationTitleAlignment ?? ContentAlignment.MiddleCenter;
         }
 
-        private void UpdateFadeValues()
-        {
-            FadeValues.FadingEnabled = _basicToastNotificationData.UseFade ?? false;
-        }
+        private void UpdateFadeValues() => FadeValues.FadingEnabled = _basicToastNotificationData.UseFade ?? false;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:64]: Why is `UseFade` nullable, just default to false !

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs [hunk 10]
```diff
@@ -87,6 +93,11 @@ private void UpdateIcon()
                     SetIcon(ToastNotificationImageResources.Toast_Notification_Hand_128_x_128);
                     break;
                 case KryptonToastNotificationIcon.SystemHand:
+#if NET8_0_OR_GREATER
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:96]: Why only scale in net8, make it consistent across all Krypton builds

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicWithProgressBarForm.cs [hunk 11]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicWithProgressBarForm.cs:1]: License, and why are these here ?

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs:101]: @Smurf-IV Implemented, but am I missing something, as it still shows up as bold?

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs (hunk 6 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs]: Now rectified

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs (hunk 7 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonMessageBox.cs:121]: Undo changes

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs (hunk 9 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs:64]: `UseFade` is now false by default

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicWithProgressBarForm.cs (hunk 11 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicWithProgressBarForm.cs:1]: Added license

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs (hunk 6 above)
Comment (Smurf-IV) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs]: You should be using the palette font for headers, and not creating a new one with a fixed size !

File: Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs (hunk 5 above)
Comment (Smurf-IV) [reply Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs:101]: You have stated `[DefaultValue(null)]` but it is a normal bool,
So needs to be set as `false` and check accordingly.
I do not see the `Should##` and `Reset###` API's

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs (hunk 6 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs]: Where is the "HeaderStyle" font stored & is there an API for retrieving it?

Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Visuals/Notification Toast/Basic/VisualToastNotificationBasicForm.cs]: Think this has now been sorted

```cs
kwlblNotificationTitle.StateCommon.Font =
                    _basicToastNotificationData.NotificationTitleFont ?? _palette.Header1ShortFont;
```

File: Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs (hunk 5 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Palette Controls/PaletteInputControlContentStates.cs:100]: This has now been fixed
